enterprise/server/sociartifactstore/sociartifactstore.go

```diff
@@ -503,7 +503,9 @@ func (s *SociArtifactStore) indexLayer(ctx context.Context, layer v1.Layer) (*re
 			layerDigest.Hex, mediaType, compressionAlgo)
 	}
 
-	layerTmpFile, err := os.Create(filepath.Join(*layerStorage, layerDigest.Hex))
+	layerTmpFileName := filepath.Join(*layerStorage, layerDigest.Hex)
+	layerTmpFile, err := os.Create(layerTmpFileName)
```

Comment:
you're better off using a random string (`random.RandomString(10)`) rather than the layer digest for the name. it's very unlikely to happen but you theoretically can get a conversion request for two images that share one or more layers which would both write to the same file.